bridge/matrix/matrix.go

```diff
@@ -75,22 +75,30 @@ func New(cfg *bridge.Config) bridge.Bridger {
 func (b *Bmatrix) Connect() error {
 	var err error
 	b.Log.Infof("Connecting %s", b.GetString("Server"))
-	b.mc, err = matrix.NewClient(b.GetString("Server"), "", "")
-	if err != nil {
-		return err
-	}
-	resp, err := b.mc.Login(&matrix.ReqLogin{
-		Type:       "m.login.password",
-		User:       b.GetString("Login"),
-		Password:   b.GetString("Password"),
-		Identifier: matrix.NewUserIdentifier(b.GetString("Login")),
-	})
-	if err != nil {
-		return err
+	if b.GetString("MxID") != "" && b.GetString("Token") != "" {
+		b.mc, err = matrix.NewClient(
+			b.GetString("Server"), b.GetString("MxID"), b.GetString("Token"),
+		)
```

Comment:
The linting problem is because you don't handle the error, you need to add an
```
if err != nil {
         return err
}
```
So that we fail when the NewClient method fails.